src/cs-core/PluginBase.cpp

```diff
@@ -10,7 +10,7 @@ namespace cs::core {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void PluginBase::setAPI(std::shared_ptr<const Settings> const& settings,
+void PluginBase::setAPI(std::shared_ptr<Settings> const& settings,
     std::shared_ptr<SolarSystem> const& solarSystem, std::shared_ptr<GuiManager> const& guiManager,
     std::shared_ptr<InputManager> const& inputManager, VistaSceneGraph* sceneGraph,
     std::shared_ptr<GraphicsEngine> const&      graphicsEngine,
```

Comment:
All of them should be passed by value and moved below.

Reply:
You're right but in order to avoid too many conflicts with the `feature/clang-tidy` I will do this after it got merged.